packages/strapi-design-system/src/CarouselInput/CarouselImage.js

```diff
@@ -1,4 +1,47 @@
-import React from 'react';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
 import { Box } from '../Box';
+import styled from 'styled-components';
+import { Tooltip } from '../Tooltip';
+import { ellipsisStyle, handleColor } from '../Typography/utils';
 
-export const CarouselImage = (props) => <Box as="img" height="100%" {...props} />;
+// ImageBase is needed since tooltip requires box level element to work upon
+// Also img component can not be directly worked with tooltip otherwise it will span the whole window
+const ImageBase = styled(Box)`
+  display: grid;
+`;
+
+const StyledImage = styled(Box)`
+  ${({ theme, color }) => `
+  width: stretch;
+  ${ellipsisStyle({ ellipsis: true })}
+  color: ${handleColor({ theme, textColor: color })};
+  `};
+`;
+
+export const CarouselImage = (props) => {
+  const [isError, setIsError] = useState(false);
+
+  const handleImageError = () => setIsError(true);
+
+  if (isError)
+    return (
+      <Tooltip description={props.alt ?? ''}>
+        <ImageBase>
+          <StyledImage as="img" {...props} onError={handleImageError} />
```

Comment:
I think this would incur an infinite loop, whereby the image you're continuously trying to load will call `setIsError` imo this handler can be omitted.

Reply:
This will not result in infinite loop since state goes from false -> false.
But I also think this is unnecessary so will remove it.